### PR TITLE
allow READ_AND_WRITE_ON_DATABASE_WHITELIST for federated metastores

### LINKED ISSUE #87: enable READ_WRITE support for federated metastores
As a user of WD
I like to run DDL on federated metastores
So that I can run ETL againt primary metastore and update federated central datalake.

Acceptance Criteria:
  ability to configure access-control-type option for federated metastores with atleast following possible values READ_ONLY, READ_WRITE.

### PR DESCRIPTION
**Pull request by courtsVII**
resolves #87

### CODE REVIEW

File: waggle-dance-api/src/test/java/com/hotels/bdp/waggledance/api/model/FederatedMetaStoreTest.java [hunk 1]
```diff
@@ -34,10 +34,12 @@ public FederatedMetaStoreTest() {
   }
 
   @Test
-  public void testReadOnlyOverride() {
+  public void testOverride() {
     assertThat(metaStore.getAccessControlType(), is(AccessControlType.READ_ONLY));
-    metaStore.setAccessControlType(AccessControlType.READ_AND_WRITE_ON_DATABASE_WHITELIST);
     assertThat(metaStore.getAccessControlType(), is(AccessControlType.READ_ONLY));
```
Comment (patduin) [inline waggle-dance-api/src/test/java/com/hotels/bdp/waggledance/api/model/FederatedMetaStoreTest.java]: this line can go I guess

File: waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/security/AccessControlHandlerFactory.java [hunk 2]
```diff
@@ -36,36 +42,39 @@ public AccessControlHandlerFactory(FederationService federationService) {
     this.federationService = federationService;
   }
 
+  private static final Logger LOG = LoggerFactory.getLogger(AccessControlHandlerFactory.class);
+
   public AccessControlHandler newInstance(AbstractMetaStore federatedMetaStore) {
     switch (federatedMetaStore.getAccessControlType()) {
     case READ_ONLY:
+      LOG.debug("Creating {} accessControlHandler for {}", READ_ONLY, federatedMetaStore.toString());
```
Comment (patduin) [inline waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/security/AccessControlHandlerFactory.java]: all these debug statements seem fairly superfluous and unnecessary, is it meaningful in the logs or can we just get rid of them

Comment (courtsVII) [reply waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/security/AccessControlHandlerFactory.java]: This actually did help me in the debugging process, I don't think they hurt and can easily make visible a configuration mistake

Comment (patduin) [reply waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/security/AccessControlHandlerFactory.java]: But they just print that READ_ONLY == READ_ONLY. 
Just add one debug line in the beginning of the method if you want to print that you're creating an accessControlHandler for a metastore.

**Pull request merged**